### PR TITLE
Improves CI

### PR DESCRIPTION
**Pull request by Elinvynia**
~~Note: This requires the GITHUB_TOKEN secret to be set in the repo settings for the pretty clippy direct lints, it's optional but nice to have.~~

### CODE REVIEW

**Comment by XAMPPRocky** [comment]
Thank you for your PR! However I would prefer to not use the `actions-rs`actions. In my opinion it's too much boilerplate now compared to using rustup directly which is always pre-installed. the clippy check would be nice but it has limitation of not working on forked repositories which makes unusable in my opinion as all pull requests are from forked repositores.

If you rewrote the clippy and format checks to use cargo directly using `run:` I would be happy to merge this in.

**Comment by Elinvynia** [comment]
> Thank you for your PR! However I would prefer to not use the `actions-rs`actions. In my opinion it's too much boilerplate now compared to using rustup directly which is always pre-installed. the clippy check would be nice but it has limitation of not working on forked repositories which makes unusable in my opinion as all pull requests are from forked repositores.
> 
> If you rewrote the clippy and format checks to use cargo directly using `run:` I would be happy to merge this in.

When was rustup added to the default installed packages? I definitely wasn't aware of this change, so thank you for bringing this to my attention, definitely gonna use it on my repos too.

**Comment by XAMPPRocky** [comment]
@Elinvynia I believe it was added at the end of last year. You can read what's exactly available here. https://help.github.com/en/actions/reference/software-installed-on-github-hosted-runners

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -6,13 +6,40 @@ on:
...
+          - type: nightly
+            rust: nightly
+            os: ubuntu-latest
+          - type: windows
+            rust: stable
+            os: windows-latest
+          - type: macos
+            rust: stable
+            os: macos-latest
+    runs-on: ${{ matrix.os }}
```
Comment (XAMPPRocky) [inline .github/workflows/rust.yml:28]: Can you either revert the change to the strategy matrix or explain the change? It doesn't look like it adds anything to me.

Comment (Elinvynia) [reply .github/workflows/rust.yml:28]: In fact it removes a tiny bit - I am open to doing it your way but this saves a little bit of processing power to only test beta and nightly on linux rather than all platforms, so shorter CI times and less runners used allowing this repo to scale better in the future at the cost of minor loss of information in case something goes wrong on these two platforms on the non-stable channel.

Comment (XAMPPRocky) [reply .github/workflows/rust.yml:28]: Yeah, I want to test beta and nightly on macos and windows too. I don't care about CI times or less runners. It takes less than 5 minutes, and I'd prefer to be able catch a upcoming breaking change or regression in Rust.

Comment (Elinvynia) [reply .github/workflows/rust.yml:28]: Alright, gonna revert it!

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -13,6 +13,18 @@ jobs:
         os: [ubuntu, macos, windows]
     steps:
     - uses: actions/checkout@v2
-    - run: rustup default ${{ matrix.channel }}
+    - run: rustup default ${{ matrix.rust }}
```
Comment (XAMPPRocky) [inline .github/workflows/rust.yml:16]:
```suggestion
    - run: rustup default ${{ matrix.channel }}
```

**Comment by XAMPPRocky** [comment]
Thank you again for your PR!
